region/maridia/inner-pink/Crab Shaft.json

```diff
@@ -154,6 +186,102 @@
       "name": "Base",
       "requires": []
     },
+    {
+      "link": [1, 2],
+      "name": "Come in Shinecharging, Leave with Shinespark (Bottom)",
+      "entranceCondition": {
+        "comeInShinecharging": {
+          "length": 1,
+          "openEnd": 0
+        }
+      },
+      "requires": [
+        "canWaterShineCharge",
+        "canHeroShot",
+        "canShinechargeMovementTricky",
+        "canMidairShinespark",
+        {"shinespark": {"frames": 39}}
+      ],
+      "exitCondition": {
+        "leaveWithSpark": {
+          "position": "bottom"
+        }
+      },
+      "unlocksDoors": [
+        {"types": ["super"], "requires": []},
+        {"types": ["missiles", "powerbomb"], "requires": ["never"]}
+      ],
+      "note": [
+        "Slide into the hole where the shot block was and sink to the bottom.",
+        "Shoot the door using a Hero shot and time the shinespark to follow the shot."
+      ]
+    },
+    {
+      "link": [1, 2],
+      "name": "Come in Shinecharging, Leave with Shinespark (Top, HiJump)",
+      "entranceCondition": {
+        "comeInShinecharging": {
+          "length": 1,
+          "openEnd": 0
+        }
+      },
+      "requires": [
+        "canWaterShineCharge",
+        "canHeroShot",
+        "canShinechargeMovementTricky",
+        "canMidairShinespark",
+        "HiJump",
+        {"or": [
+          {"and": [
+            "Plasma",
+            "Wave"
+          ]},
+          "canStationarySpinJump"
+        ]},
+        {"shinespark": {"frames": 41}}
+      ],
+      "exitCondition": {
+        "leaveWithSpark": {}
+      },
+      "unlocksDoors": [
+        {"types": ["super"], "requires": [{"ammo": {"type": "Super", "count": 1}}]},
+        {"types": ["missiles", "powerbomb"], "requires": ["never"]}
+      ],
+      "note": [
+        "Slide into the hole where the shot block was and sink to the bottom.",
+        "Shoot the door using a Hero shot and time the shinespark to follow the shot.",
+        "Either kill the ceiling crabso that Samus can jump forward, or stationary spinjump to avoid using the Shinespark early."
```

Comment:
```suggestion
        "Either kill the ceiling crab so that Samus can jump forward, or stationary spinjump to avoid using the Shinespark early."
```